### ISSUE TITLE
[Bug]: Killed when quantizing 1020B model

### ISSUE THREAD

**Issue by IEI-mjx** [vllm-project/llm-compressor#2143]
### ⚙️ Your current environment

I successfully quantize our 40B model, but failed when quantizing 1020B model.
CPU mem: 2TB
GPU: A800 8*80GB 

### 🐛 Describe the bug

Here is the log:
<img width="1424" height="128" alt="Image" src="https://github.com/user-attachments/assets/e38b87e6-5e31-4811-a3d1-6dd8b6a8e50d" />

### 🛠️ Steps to reproduce

Here is our script:
```

import requests
import torch
from PIL import Image
from transformers import AutoProcessor, AutoModel, AutoTokenizer

from llmcompressor import oneshot
from llmcompressor.modifiers.quantization import GPTQModifier
from llmcompressor.utils import dispatch_for_generation
from datasets import load_dataset
import os
import glob

MAX_SEQUENCE_LENGTH = 32768
model_id = "/mnt/beegfs/maojunxiong/yuanvl_1020B_0002350"
model = AutoModel.from_pretrained(model_id, torch_dtype="auto", trust_remote_code=True,  device_map="auto", use_flash_attn=False)
tokenizer = AutoTokenizer.from_pretrained(model_id, trust_remote_code=True)

num_token_per_tile = int(args.clip_visual_size * args.downsample_ratio**2)


image_dir = '/mnt/beegfs/maojunxiong/image'
image_paths = []
for ext in ['*.png', '*.jpg', '*.jpeg']:
    image_paths.extend(glob.glob(os.path.join(image_dir, ext)))

image_paths = image_paths[:512]
questions = ['Please describe the picture'] * len(image_paths)
all_datasets = []
for img_path, q in zip(image_paths, questions):
    prompts, images_input = preocess_imput(args, num_token_per_tile, [img_path], [q])

    input = tokenizer(prompts, return_tensors="pt")
    input_ids = input['input_ids']
    pixel_values = images_input['image_tensor']

    if pixel_values.dim() == 4:
        if pixel_values.size(0) == 1:
            pixel_values = pixel_values.squeeze(0)
        else:
            pixel_values = pixel_values[0]

    attention_mask, position_ids, image_info = _build_yuanvl_attention_mask_and_position_ids(
                    tokenizer, input_ids, images_input)

    attention_mask = input['attention_mask']

    
    input_ids = input_ids.cpu().float() if input_ids.dtype == torch.bfloat16 else input_ids.cpu()
    attention_mask = attention_mask.cpu().float() if attention_mask.dtype == torch.bfloat16 else attention_mask.cpu()
    position_ids = position_ids.cpu().float() if position_ids.dtype == torch.bfloat16 else position_ids.cpu()
    pixel_values = pixel_values.cpu().float() if pixel_values.dtype == torch.bfloat16 else pixel_values.cpu()

    dataset_dict = {
        "input_ids": input_ids.numpy(),
        "attention_mask": attention_mask.numpy(),
        "position_ids": position_ids.numpy(),
        "pixel_values": pixel_values.numpy()
    }

    from datasets import Dataset
    single_dataset = Dataset.from_dict(dataset_dict)
    all_datasets.append(single_dataset)

from datasets import concatenate_datasets
dataset = concatenate_datasets(all_datasets)
# Recipe
recipe = [
    GPTQModifier(
        targets="Linear",
        scheme="W4A16",
        ignore=["re:.*lm_head$", "re:.*qkv$", "re:.*fc1$", "re:.*fc2$",  "re:.*attn.proj$", "re:.*up_proj$", "re:.*gate_proj$", "re:.*down_proj$", "re:.*router.query_key_value$"]
    ),
]


oneshot(
    model=model,
    tokenizer=model_id,
    dataset=dataset,
    recipe=recipe,
    max_seq_length=MAX_SEQUENCE_LENGTH,
    #num_calibration_samples=NUM_CALIBRATION_SAMPLES,
    trust_remote_code_model=True,
    tie_word_embeddings=True,
    sequential_targets=["YuanDecoderLayer"],
)

SAVE_DIR = "/mnt/beegfs/maojunxiong/yuanvl_1020B_0002350-int4"
model.save_pretrained(SAVE_DIR, save_compressed=True)
tokenizer.save_pretrained(SAVE_DIR)

```

**Comment by brian-dellabetta** [comment]
Hi @IEI-mjx , we hope to land some enhancements early next year to help with quantizing super large models like kimi k2 and deepseek, some more information can be found in the thread at #1650 

In the meantime, have you tried data-free flows just using QuantizationModifier? You could try `FP8_DYNAMIC`, `W4A16`, and `W4A16_ASYM` schemes. You might be able to use them with minimal degradation to accuracy on very large models, for example our published deepseek w4a16 model has really good recovery -- https://huggingface.co/RedHatAI/DeepSeek-R1-quantized.w4a16

**Comment by IEI-mjx** [comment]
Thanks for your repo!!
I will try it later.
By the way, I have one more question. I quantized our 1020B model using llm-compressor-0.2.0 with
 `device_map = custom_offload_device_map(MODEL_ID, max_memory_per_gpu="20GB", num_gpus=8, torch_dtype=torch.bfloat16)`
which was deleted in latest version. This maybe helpful for super large models.

**Comment by dsikka** [comment]
Hi @IEI-mjx 
Do you mind setting device_map=None or not setting it at all and trying again? We will handle the memory management for you; https://github.com/vllm-project/llm-compressor/blob/main/examples/quantization_w4a16/llama3_example.py is a good example to consider

**Comment by IEI-mjx** [comment]
> Hi [@IEI-mjx](https://github.com/IEI-mjx) Do you mind setting device_map=None or not setting it at all and trying again? We will handle the memory management for you; https://github.com/vllm-project/llm-compressor/blob/main/examples/quantization_w4a16/llama3_example.py is a good example to consider

I set `device_map=None` and add one more argument  `pipeline=layer_sequential` in oneshot(), which works for me!

**Comment by IEI-mjx** [comment]
Hi @dsikka 
Using these args, the first layer of our model could be quantized. But went wrong at the second layer when Calibrating.

<img width="1182" height="546" alt="Image" src="https://github.com/user-attachments/assets/a952985d-ed70-4190-99a1-d5f0bfe88bd7" />

What is the difference between sequential and layer_sequential?

> > Hi [@IEI-mjx](https://github.com/IEI-mjx) Do you mind setting device_map=None or not setting it at all and trying again? We will handle the memory management for you; https://github.com/vllm-project/llm-compressor/blob/main/examples/quantization_w4a16/llama3_example.py is a good example to consider
> 
> I set `device_map=None` and add one more argument `pipeline=layer_sequential` in oneshot(), which works for me!

**Comment by brian-dellabetta** [comment]
Hi @IEI-mjx , we deprecated layer_sequential, `pipeline="sequential"` is the right one to use. Can you confirm you are on latest llm-compressor? 

This might be something model-specific, but you can try a data-free pathway as a first sanity check

**Comment by IEI-mjx** [comment]
> Hi [@IEI-mjx](https://github.com/IEI-mjx) , we deprecated layer_sequential, `pipeline="sequential"` is the right one to use. Can you confirm you are on latest llm-compressor?
> 
> This might be something model-specific, but you can try a data-free pathway as a first sanity check

I'm using the llm-compressor-v0.7.1.
As far as i know, layer_sequential is suitable for models that infer without cache. Our visionmodel have both kv-cache and convolution layer cache, so only sequential can be used.
I have tested several cases in our 40B model with GPTQmodifier and the results show that the num of calibration data  will influence the accuracy heavily! 
Now we have 8xH800 nodes, the CPU memory is 2T, our largest model size is 1020B (weight size 2T). Do you have any solution for this? Like load some weight on GPU(we have 8x80GB memory)? Or using 2 nodes?

**Comment by brian-dellabetta** [comment]
Hi @IEI-mjx , few thougts:

1. strongly recommend upgrading to latest llm-compressor v0.9.0. LayerSequentialPipeline will have been removed in latest, the SequentialPipeline should be fine. 
2. I don't think you want to run with caching during calibration, it has the possibility of pulling cached values from the unquantized model when you really want to forward pass through to get the outputs if things were quantized. We generally disable it during the modifiers anyway, example [here](https://github.com/vllm-project/llm-compressor/blob/main/src/llmcompressor/modifiers/awq/base.py#L411-L415) with [calibration_forward_context](https://github.com/vllm-project/llm-compressor/blob/main/src/llmcompressor/utils/helpers.py#L1077-L1093)
3. While calibration might be important for 40B, it might be way less important for such a large model (as shown in the model card for our deepseek r1 w4a16 checkpoint i posted above). You can experiment with it and see how it goes.
4. Barring any of that, we plan to improve support for calibrated large model compression early next year, as well as multi-GPU. Stay tuned!

**Comment by IEI-mjx** [comment]
Thanks a lot, @brian-dellabetta 
Could you please give me an example of `deepseek r1 w4a16` quantization script?
I tested `pipeline=datafree`  to quantize our 1000B model and got no content from the output of the w4a16 model.
Below is our script:
```

MAX_SEQUENCE_LENGTH = 32768

import requests
import torch
from PIL import Image
from transformers import AutoProcessor, AutoModel, AutoTokenizer

from llmcompressor import oneshot
from llmcompressor.modifiers.quantization import GPTQModifier
from llmcompressor.utils import dispatch_for_generation
from datasets import load_dataset
import os
import glob


model_id = "/mnt/beegfs/maojunxiong/yuanvl_1020B_0002350"

model = AutoModel.from_pretrained(model_id, torch_dtype="auto", trust_remote_code=True, use_flash_attn=False)
tokenizer = AutoTokenizer.from_pretrained(model_id, trust_remote_code=True)
recipe = [
    GPTQModifier(
        targets="Linear",
        scheme="W4A16",
        ignore=["re:.*lm_head$", "re:.*qkv$", "re:.*fc1$", "re:.*fc2$",  "re:.*attn.proj$", "re:.*up_proj$", "re:.*gate_proj$", "re:.*down_proj$", "re:.*router$"]
    ),
]
oneshot(
    model=model,
    tokenizer=model_id,
    recipe=recipe,
    max_seq_length=MAX_SEQUENCE_LENGTH,
    trust_remote_code_model=True,
    pipeline="datafree",
    tie_word_embeddings=True,
    sequential_targets=["YuanDecoderLayer"],
)

SAVE_DIR = "/mnt/beegfs/maojunxiong/yuanvl_1020b_0002350-int4"
model.save_pretrained(SAVE_DIR, save_compressed=True)
tokenizer.save_pretrained(SAVE_DIR)
```

**Comment by brian-dellabetta** [comment]
GPTQ is not a data-free algorithm. use `QuantizationModifier` instead of `GPTQModifier`, for example how it's set up on [this line](https://github.com/vllm-project/llm-compressor/blob/c905a04eb28ff5aee705aad9b998ea64dd747534/examples/transform/spinquant_example.py#L25)

**Comment by IEI-mjx** [comment]
> GPTQ is not a data-free algorithm. use `QuantizationModifier` instead of `GPTQModifier`, for example how it's set up on [this line](https://github.com/vllm-project/llm-compressor/blob/c905a04eb28ff5aee705aad9b998ea64dd747534/examples/transform/spinquant_example.py#L25)

Thanks, i will try it later.
By the way, does `QuantizationModifier` support `int4 W4A16` ? 
And for `GPTQModifier`, we expand our CPU memory from 2T to 4T, maybe this will be helpful for `GPTQ-int4 quantization`?

**Comment by brian-dellabetta** [comment]
@IEI-mjx  yes QuantizationModifier supports W4A16 data-free. Increasing to 4T may help, but I would suspect you could still run into OOM errors for a calibrated compression of a 1020B model. It depends on how large of a dataset you are using

**Comment by IEI-mjx** [comment]
It works for me to quantize 1020B model using GPTQModifier with 256 calibration samples after increasing cpu memory to 4T.

**Comment by brian-dellabetta** [comment]
Thanks @IEI-mjx for the information!
